Ghidra/Processors/CP1600/data/languages/CP1600.slaspec

```diff
@@ -0,0 +1,570 @@
+cc: "C"   is branch_external=0 & branch_condition=1  { export C; }
+cc: "OV"  is branch_external=0 & branch_condition=2  { export O; }
+cc: "PL"  is branch_external=0 & branch_condition=3  { local tmp = !S; export tmp; }
+cc: "EQ"  is branch_external=0 & branch_condition=4  { export Z; }
+cc: "LT"  is branch_external=0 & branch_condition=5  { local tmp = S != O; export tmp; }
+cc: "LE"  is branch_external=0 & branch_condition=6  { local tmp = (Z == 1) || (S != O); export tmp; }
+cc: "USC" is branch_external=0 & branch_condition=7  { local tmp = S != C; export tmp; }
+cc: "NC"  is branch_external=0 & branch_condition=9  { local tmp = !C; export tmp; }
+cc: "NOV" is branch_external=0 & branch_condition=10 { local tmp = !O; export tmp; }
+cc: "MI"  is branch_external=0 & branch_condition=11 { export S; }
+cc: "NEQ" is branch_external=0 & branch_condition=12 { local tmp = !Z; export tmp; }
+cc: "GE"  is branch_external=0 & branch_condition=13 { local tmp = S == O; export tmp; }
+cc: "GT"  is branch_external=0 & branch_condition=14 { local tmp = (Z == 0) || (S == O); export tmp; }
+cc: "ESC" is branch_external=0 & branch_condition=15 { local tmp = S == C; export tmp; }
+
+################################################################
+
+macro resultFlags(value) {
+	Z = value == 0;
+	S = value s< 0; 
+}
+
+macro addition(first, second) {
+	local tmpC = carry(first, second);
+	local tmpO = scarry(first, second);
+	first = first + second;
+	C = tmpC;
+	O = tmpO;
+	resultFlags(first);
+}
+
+macro comparison(first, second) {
+	local __val__ = first - second;
+	O = sborrow(first, second);
+	C = first < second; 
+	resultFlags(__val__);
+}
+
+macro subtraction(first, second) {
+	local __val__ = first - second;
+	O = sborrow(first, second);
+	C = first < second; 
+	resultFlags(__val__);
+	first = __val__;
+}
+
+################################################################
+
+define pcodeop TerminateCurrentInterrupt;
+define pcodeop SoftwareInterrupt;
+
+################################################################
+
+:ADD addr16, reg0_2 is opcode3_9=0x0058 & reg0_2 ; addr16 {
+	local ptr:2 = addr16;
+	addition(reg0_2, *:2 ptr);
+}
+
+:ADD@ impliedval16, reg0_2 is opcode6_9=0x000B & reg0_2 & impliedval16 {
+	addition(reg0_2, impliedval16);
+}
+
+:ADCR reg0_2 is opcode3_9=0x0005 & reg0_2 {
+	local oldC = zext(C);
+	addition(reg0_2, oldC);
+}
+
+:ADDR reg3_5, reg0_2 is opcode6_9=0x0003 & reg3_5 & reg0_2 {
+	addition(reg0_2, reg3_5);
+}
+
+:AND addr16, reg0_2 is opcode3_9=0x0070 & reg0_2 ; addr16 {
+	local ptr:2 = addr16;
+	reg0_2 = reg0_2 & *:2 ptr;
+	resultFlags(reg0_2);
+}
+
+:AND@ impliedval16, reg0_2 is opcode6_9=0x000E & reg0_2 & impliedval16 {
+	reg0_2 = reg0_2 & impliedval16;
+	resultFlags(reg0_2);
+}
+
+:ANDR reg3_5, reg0_2 is opcode6_9=0x0006 & reg3_5 & reg0_2 {
+	reg0_2 = reg0_2 & reg3_5;
+	resultFlags(reg0_2);
+}
+
+:B^cc branchdest16 is (opcode6_9=0x0008 & cc) ... & branchdest16 {
+	if (!cc) goto inst_next;
```

Comment:
I would remove the extra goto and just do:  `if (cc) goto branchdest16;`

Reply:
Oops, brain fart. I'll sort it out ASAP.